lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -577,464 +665,528 @@
-                return __fromNLS.get(locale);
-            }
+        const InitializeCollator = function (collator, locales, options) {
+            const requestedLocales = CanonicalizeLocaleList(locales);
+            options = options === undefined ? _.create() : Internal.ToObject(options);
 
-            const parts = platform.builtInRegexMatch(locale, /([^_]*)_?(.+)?/);
-            const strippedLocale = parts[1];
-            const collation = parts[2];
-            let ret;
+            collator.usage = GetOption(options, "usage", "string", ["sort", "search"], "sort");
+            const collatorLocaleData = collator.usage === "sort" ? localeData : localeData; // TODO
 
-            if (collation === undefined) {
-                return locale;
-            }
+            const opt = _.create();
+            opt.matcher = GetOption(options, "localeMatcher", "string", ["lookup", "best fit"], "best fit");
+            let kn = GetOption(options, "numeric", "boolean", undefined, undefined);
+            opt.kn = kn === undefined ? kn : Internal.ToString(kn);
+            opt.kf = GetOption(options, "caseFirst", "string", ["upper", "lower", "false"], undefined);
+            
+            const r = ResolveLocale(undefined, requestedLocales, opt, ["co", "kn", "kf"]);
+            collator.locale = r.locale;
+            collator.collation = r.co === null ? "default" : r.co;
+            collator.numeric = r.kn === "true";
+            collator.caseFirst = r.kf;
+            collator.caseFirstEnum = toEnum(CollatorCaseFirst, collator.caseFirst);
 
-            const collationMapForLocale = reverseLocalesAcceptingCollationValues[strippedLocale];
-            if (collationMapForLocale === undefined) {
-                // TODO(jahorto): NLS gave us a locale_collation back but we don't have a map for that locale -- why?
-                ret = `${strippedLocale}-u-co-${collation}`;
-                __fromNLS.set(locale, ret);
-                return ret;
-            }
+            collator.sensitivity = GetOption(options, "sensitivity", "string", ["base", "accent", "case", "variant"], "variant");
+            collator.sensitivityEnum = toEnum(CollatorSensitivity, collator.sensitivity);
+    
+            collator.ignorePunctuation = GetOption(options, "ignorePunctuation", "boolean", undefined, false);
 
-            const mappedCollation = collationMapForLocale[collation];
-            if (mappedCollation !== undefined) {
-                ret = `${strippedLocale}-u-co-${mappedCollation}`;
-            } else {
-                // TODO(jahorto): NLS gave us a locale_collation back but we dont think that collation is supported for that locale -- why?
-                ret = `${strippedLocale}-u-co-${collation}`;
+            collator.initializedCollator = true;
+
+            return collator;
+        };
+
+        platform.registerBuiltInFunction(tagPublicFunction("String.prototype.localeCompare", function () {
```

Comment:
Shouldn't this take the arguments mentioned in the spec, e.g. `localeCompare ( that [ , locales [ , options ] ] )`?

Reply:
Will fix in the next PR